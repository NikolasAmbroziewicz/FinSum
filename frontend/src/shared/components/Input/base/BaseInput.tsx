import { ChangeEventHandler } from 'react';
import { UseFormRegisterReturn } from 'react-hook-form';

export interface IBaseInput {
  id: string;
  placeholder: string;
  type: string;
  changeHandler?: ChangeEventHandler;
  formHandler?: UseFormRegisterReturn;
  value?: string;
  error?: boolean;
  step?: string,
}

const BaseInput: React.FC<IBaseInput> = ({
  id,
  placeholder,
  type,
  changeHandler,
  formHandler,
  value,
  error,
  step = '1'
}) => {
  const borderClass = () => {
    return error ? 'border-rose-400' : 'border-slate-300';
  };

  return (
    <>
      <input
        className={`w-full shadow border ${borderClass()} rounded py-2 px-3 text-gray-700 leading-tight 
          focus:outline-none focus:shadow-outline focus:borderfocus:border-sky-600
          placeholder:text-slate-300
          `}
        id={id}
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={changeHandler}
        step={step}
        min={0}
        {...formHandler}
      />
    </>
  );
};

export default BaseInput;
