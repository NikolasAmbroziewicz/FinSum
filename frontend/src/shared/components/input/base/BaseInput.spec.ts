import { describe, expect, it } from 'vitest';
import BaseInput from './BaseInput';
import { mockBaseInputProps } from './BaseInput.mocks';

describe('BaseInput', () => {
  it('defined', () => {
    expect(BaseInput(mockBaseInputProps.base)).toMatchSnapshot();
  });
});
