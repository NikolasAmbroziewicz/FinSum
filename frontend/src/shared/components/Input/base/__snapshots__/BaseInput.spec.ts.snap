// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > defined 1`] = `
<React.Fragment>
  <input
    className="w-full shadow border border-slate-300 rounded py-2 px-3 text-gray-700 leading-tight 
          focus:outline-none focus:shadow-outline focus:borderfocus:border-sky-600
          placeholder:text-slate-300
          "
    id="Base Input"
    placeholder="Base Input"
    type="text"
  />
</React.Fragment>
`;
